testing/dart/window_hooks_integration_test.dart

```diff
@@ -55,55 +57,68 @@ void main() {
       window.onSemanticsEnabledChanged = originalOnSemanticsEnabledChanged;
       window.onSemanticsAction = originalOnSemanticsAction;
       window.onPlatformMessage = originalOnPlatformMessage;
+      window.onTextScaleFactorChanged = originalOnTextScaleFactorChanged;
     });
 
     test('onMetricsChanged preserves callback zone', () {
       Zone innerZone;
       Zone runZone;
+      double devicePixelRatio;
 
       runZoned(() {
         innerZone = Zone.current;
         window.onMetricsChanged = () {
           runZone = Zone.current;
+          devicePixelRatio = window.devicePixelRatio;
         };
       });
 
       window.onMetricsChanged();
-      _updateWindowMetrics(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0);
+      _updateWindowMetrics(0.1234, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0);
       expect(runZone, isNotNull);
       expect(runZone, same(innerZone));
+      expect(devicePixelRatio, isNotNull);
+      expect(devicePixelRatio, equals(0.1234));
```

Comment:
Isn't equals(1234) implying that it cannot be null?

Reply:
see also a couple of times further down.

Reply:
Yeah, I guess you're right.  I add those because when the test fails because it's null, I thought it would be nicer to have it highlight that it was null, but I guess it's pretty obvious for the equals case.